### PR TITLE
Add analysis code for several detectors

### PR DESCRIPTION
**Pull request by jang00777**
1. ME0 / GEM / CSC / DT / RPC
    - digi / rechit / segment
2. h_counts and h_area for plotting hit rate comparison plot

### CODE REVIEW

File: MuonAnalyser/test/gemsim_on_condor.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh
+
+# Steps from "runTheMatrix.py -w upgrade -l 10811.0"
+
+cd /cms/scratch/iwatson/GEM/CMSSW_10_3_0_pre3_vanilla/src/GEMSimulation
```
Comment (jshlee) [inline MuonAnalyser/test/gemsim_on_condor.sh]: why are you using this?

**Comment by jshlee** [comment]
why are there slice test stuff in this PR?

try to make better names i.e. instead of HGCalSimTest.cc have MuonDetHitAnalzser.cc and MuonTrackAnalyzer.cc

File: MuonAnalyser/plugins/HGCalSimTest.cc [hunk 2]
```diff
@@ -0,0 +1,1072 @@
...
+  edm::Handle<reco::VertexCollection> vertices;
+  iEvent.getByToken(vtxToken_, vertices); 
+
+  vertexes_ = vertices.product();
+  reco::Vertex pv0 = vertexes_->at(0);
+
+  initValue();
+
+  hEvents_->Fill(1);
+
```
Comment (jshlee) [inline MuonAnalyser/plugins/HGCalSimTest.cc]: i think its better to loop over digis and rechits instead,

File: MuonAnalyser/plugins/SliceTestAnalysis.cc [hunk 3]
```diff
@@ -1,545 +0,0 @@
-// cd /cms/ldap_home/iawatson/scratch/GEM/CMSSW_10_1_5/src/ && eval `scramv1 runtime -sh` && eval `scramv1 runtime -sh` && scram b -j 10
```
Comment (jshlee) [inline MuonAnalyser/plugins/SliceTestAnalysis.cc:1]: i dont think this should be deleted. doesnt @tt8888tt use this?

Comment (watson-ij) [reply MuonAnalyser/plugins/SliceTestAnalysis.cc:1]: I think its from one of my codes from a long time ago. Im pretty sure DJ can't compile in my home directory

Comment (tt8888tt) [reply MuonAnalyser/plugins/SliceTestAnalysis.cc:1]: Yep Ian is right

**Pull request merged**